lib/api/repository/survey_repository.dart

```diff
@@ -24,7 +26,10 @@ class SurveyRepositoryImpl extends SurveyRepository {
     try {
       final response = await _surveyService.getSurveys(pageNumber, pageSize);
       final surveys = response.surveys ?? [];
-      return surveys.map((item) => SurveyModel.fromResponse(item)).toList();
+      final surveyModels =
+          surveys.map((item) => SurveyModel.fromResponse(item)).toList();
+      _surveyStorage.saveSurveys(surveyModels);
```

Comment:
Does that mean we will override the cache when fetching the second page?

Reply:
Since loading more pages wasn't mentioned in the [requirements](https://www.notion.so/nimblehq/Internal-Certification-Path-Mobile-de354e1ce76f4b80922c3db21ea0117f), we will not implement this feature 🙏